tests/gordo_components/server/test_base_view.py

```diff
@@ -69,19 +73,23 @@ def test_prediction_endpoint_post_ok(
 
 
 @pytest.mark.parametrize(
-    "data_to_post",
+    "data_to_post,error_generator",
     [
-        {"X": [list(range(len(tu.SENSORTAG_LIST) - 1))]},  # Not enough features
-        {"no-x-here": True},  # No X
-        {"X": [[1, 2, [1, 2]]]},  # Badly formatted data
+        # Not enough features
+        lambda X: X[:-1, :],
+        # No X
+        lambda X: True,
+        # Badly formatted data
+        lambda X: [[1, 2, [1, 2]]],
     ],
 )
 def test_prediction_endpoint_post_fail(
-    caplog, base_route, sensors, gordo_ml_server_client, data_to_post
+    caplog, base_route, sensors, gordo_ml_server_client, error_generator
 ):
     """
     Test expected failures when posting certain types of data
     """
+    data_to_post = {"X": error_generator(np.random.random((len(sensors), 10)))}
```

Comment:
I find this harder to read, and it takes more lines..

Reply:
I liked it a bit better before too. This was done to address not being able to pass fixtures to the `parametrize` decorater.

Reply:
My thinking here was that we'd start with a valid `X` and some mutation to it yields the erroneous data being tested.

Reply:
IMO it 'breaks' part of the test, for example 'No X' had a _key_ other than `X` now it puts the _value_ as `True`. I still don't see how this is necessary.

Reply:
This is necessary (but could of course be wrong here) because I can no longer call the global `tu.SENSORS_STR_LIST` to the `pytest.mark.parametrize` decorator.  I need to now pass the `sensors_str` fixture to the test, and then generate the data in the test. 

I can fix the key bit easily though.

Reply:
you don't _need_ to have `tu.SENSORS_STR_LIST` though, right? I mean, having an empty list is clearly going to be 'not enough features' or even make it 10,000 features, because we surely won't have that either.

Reply:
Ok, have a simple solution that just omits the parametrize...

Reply:
Cool, LGTM, I'll approve once tests are passing. :+1: